pkg/status/appregistryerror.go

```diff
@@ -0,0 +1,40 @@
+package status
+
+// opsrcErrorConst implements errors related to default OperatorSources.
+type opsrcErrorString string
+
+const (
+	AppRegistryListPackagesError  opsrcErrorString = "AppRegistryListPackagesError"
```

Comment:
It seems like you defined all of these types, but they're not used in the control flow at all. If that's the case, why not just craft the strings inline? I thought the purpose of the custom error type here was to differentiate between actual errors from the app-registry API and construction errors. If we are defining them to mean the same thing (from the perspective of the status reporting) then what value do they bring other than having different error strings?

Reply:
Crafting the strings in line is a good proposal. The ClusterOperator allows us to report a `Message` and a `Reason`, where:

- The `Message` can be a string
- The `Reason` should be CamelCase.

Given that we want to use both, I suggest keeping the data structure but removing the `consts` as you suggested.

Reply:
Also - `Reason` is available via telemetry where `Message` is not.